캠핑용품 타입 목록 조회 기능 개발

Habist opened the pull request:
## 작업 내용
캠핑용품 타입 목록을 DB로 부터 조회하는 기능을 개발하였습니다.

## 주의 사항


>  참조이슈
> #26

Habist committed:
feat: [#26] 캠핑용품 타입 목록 조회 기능 개발

캠핑용품 타입 목록을 조회하는 기능을 추가하였습니다.

f-lab-dev commented on src/main/java/com/campool/mapper/GearTypeMapper.java:
```diff
@@ -0,0 +1,14 @@
+@Mapper
+public interface GearTypeMapper {
+
+    @Select("SELECT ID, NAME FROM TYPE")
```
어떻게 보면 비효율적일 수도 있는 쿼리를 작성할때는 작성의도를 주석에 적어주시면 좋을 것 같습니다~

f-lab-dev commented on src/main/java/com/campool/model/GearType.java:
```diff
@@ -0,0 +1,15 @@
+import lombok.ToString;
+
+@Getter
+@Setter
```
이거는 setter가 필요한가요?

Habist replied:
이 부분은 제 실수네요. 처음에 개발할 때 setter가 있어야 동작했었는데  MyBatis 문서를 다시 확인해보니 불변 도메인 클래스도 바인딩을 지원해준다고 나와있고 실제 테스트 해보니 잘 동작하네요! 다른 클래스도 변경하였습니다~

Habist committed:
refactor: [#26] 모델 클래스 setter 제거

모델 클래스의 setter를 제거하여 불변 속성을 가지도록 변경하였습니다.

Habist committed:
refactor: [#26] selectGearTypes 주석문 추가

해당 쿼리에 대한 설명을 나타내는 주석문을 추가하였습니다.

f-lab-dev commented on src/main/java/com/campool/mapper/GearTypeMapper.java:
```diff
@@ -8,6 +8,9 @@
 
+    /**
+     * 렌트정보 조회 시 참조하는 캠핑용품 전체 유형을 가져오는 쿼리
+     */
```
아 이거는 의도를 적어달라는 이유를 다시 설명드리자면.
다른 개발자가 이 쿼리를 봤을때 `왜 테이블 전체를 가져오는 쿼리를 써놓은거야? 성능에 안 좋아보이는데?` 라고 생각할 수 있습니다.
즉 데이터가 그렇게 많지 않기 때문에 전체 캐싱을 위해 테이블 전체를 가져온다라는 이유를 적어달라는 것이였습니다~

Habist replied:
아하! 넵 리뷰 반영했습니다~

Habist committed:
refactor: [#26] 캠핑용품 유형 데이터 조회 쿼리 주석문 수정

Habist committed:
refactor: [#26] 캠핑용품 유형 데이터 조회 쿼리 주석문 수정

Habist committed:
위치 기반 렌트정보 조회 시 타입 항목 추가 (#30)

* improve: [#27] 캠핑용품 등록 시 타입 항목 id로 입력되도록 변경

* feat: [#27] 렌트정보 조회 시 타입 검색에 대한 조건 추가

위치 기반으로 렌트정보 조회 시 캠핑용품 중 타입에 대한 검색 조건을 줄 수 있도록 추가하였습니다.

* refactor: [#27 - 리뷰 반영] RentalsRequestByLocation typeId Nullable 애노테이션 추가

요청 시 Nullable한 필드에 명확한 표현을 위해 애노테이션을 추가하였습니다.

* 렌트정보 상세보기 기능 개발 (#31)

렌트 ID를 기반으로 상세정보와 해당 렌트의 캠핑용품 목록을 가져오는 기능 개발
